### PR TITLE
Move most of VersionUtility to VersionUtility2 in Tooling

### PR DESCRIPTION
**Pull request by davidfowl**
- Moved all of the compatibility checks
- Moved all of the portable framework parsing code
- Removed support for portable-x+y+z in project.json. Only
profile fully spelled out portable profiles are supported.

### CODE REVIEW

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/VersionUtility.cs [hunk 1]
```diff
@@ -524,36 +496,21 @@ public static string GetShortFrameworkName(FrameworkName frameworkName)
...
-                if (portableProfile != null)
-                {
-                    profile = portableProfile.CustomProfileString;
-                }
-                else
-                {
-                    profile = frameworkName.Profile;
-                }
+                // Remove the . from versions
+                name += frameworkName.Version.ToString().Replace(".", String.Empty);
```
Comment (muratg) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/VersionUtility.cs]: Why?

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/VersionUtility.cs [hunk 2]
```diff
@@ -599,6 +556,19 @@ public static FrameworkName ParseFrameworkNameFromFilePath(string filePath, out
             return null;
         }
 
+        public static bool IsPortableFramework(this FrameworkName framework)
```
Comment (muratg) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/VersionUtility.cs]: Can't this be a property on FrameworkName?

File: src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+        }
+
+        public static bool GetNearest<T>(FrameworkName projectFramework, IEnumerable<T> items, out IEnumerable<T> compatibleItems) where T : IFrameworkTargetable
+        {
+            return GetNearestCore(projectFramework, items, i => i.SupportedFrameworks, out compatibleItems);
+        }
+
+        private static bool GetNearestCore<T>(FrameworkName projectFramework, IEnumerable<T> items, Func<T, IEnumerable<FrameworkName>> fxExtractor, out IEnumerable<T> compatibleItems)
+        {
+            if (!items.Any())
```
Comment (muratg) [inline src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs]: Can `items` be `Null`?

File: src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+                return true;
+            }
+
+            // Not all projects have a framework, we need to consider those projects.
+            var internalProjectFramework = projectFramework ?? VersionUtility.EmptyFramework;
+
+            // Turn something that looks like this:
+            // item -> [Framework1, Framework2, Framework3] into
+            // [{item, Framework1}, {item, Framework2}, {item, Framework3}]
+            var normalizedItems = from item in items
```
Comment (muratg) [inline src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs]: Nit: formatting looks weird.

File: src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+                }
+            }
+
+            // Try the old way
+            return TryGetCompatibleItemsCore(out compatibleItems, internalProjectFramework, frameworkGroups);
+        }
+
+        private static bool TryGetCompatibleItems<T>(FrameworkName projectFramework, IEnumerable<T> items, out IEnumerable<T> compatibleItems) where T : IFrameworkTargetable
+        {
+            if (!items.Any())
```
Comment (muratg) [inline src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs]: null?

File: src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs (hunk 3 above)
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/NuGet/Utility/VersionUtility2.cs]: No, also this isn't new code.

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/VersionUtility.cs (hunk 2 above)
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/VersionUtility.cs]: No, FrameworkName is a .NET type.

**Comment by anurse** [comment]
NuGet3 added a bunch of embedded data, maybe just do that?

**Comment by davidfowl** [comment]
@anurse yah, I think I will. One question is how do we map xamarin? Do we assume that profiles map to the largest set of platforms? 

/cc @emgarten

**Comment by anurse** [comment]
Xamarin has pretty well-defined mappings as well I believe.

**Comment by emgarten** [comment]
Xamarin frameworks map to the most common portable profiles, some of the rare profiles don't have them.

**Comment by davidfowl** [comment]
This change needs a do-over

**Comment by anurse** [comment]
Wimp. You could've resolved the conflicts 😜

**Pull request closed without merging**